### PR TITLE
Fix flaky unit test failures

### PR DESCRIPTION
**Pull request by MatkovIvan**

### CODE REVIEW

File: AppCenterCrashes/AppCenterCrashesTests/MSErrorLogFormatterTests.mm [hunk 1]
```diff
@@ -164,12 +164,15 @@ - (void)testProcessIdAndExceptionForObjectiveCExceptionCrash {
 }
 
 - (void)testSelectorForRegisterWithName {
-  NSData *crashData = [[[MSCrashes sharedInstance] plCrashReporter] generateLiveReport];
+
```
Comment (TroubleMakerBen) [inline AppCenterCrashes/AppCenterCrashesTests/MSErrorLogFormatterTests.mm]: Does switching this to a fixture help with improving the flakiness of this test? This is interesting!

Comment (TroubleMakerBen) [reply AppCenterCrashes/AppCenterCrashesTests/MSErrorLogFormatterTests.mm]: On the other hand, the `generateLiveReport`-API doesn't generate a "full" report but a "live crash report" that's non-crashing, so probably not the way to unit test the selectors anyway. ^^

File: AppCenterDistribute/AppCenterDistributeTests/MSDistributeTests.m [hunk 2]
```diff
@@ -112,6 +116,7 @@ - (void)tearDown {
   [self waitForExpectations:@[ expection ] timeout:1];
 
   // Clear
+  [OHHTTPStubs removeAllStubs];
```
Comment (TroubleMakerBen) [inline AppCenterDistribute/AppCenterDistributeTests/MSDistributeTests.m]: Just nit-picking. If we have `MSHttpTestUtil`, why not move this line over there, too? I know that it is just one line, but it seems to be more concise to me.

File: AppCenterCrashes/AppCenterCrashesTests/MSErrorLogFormatterTests.mm (hunk 1 above)
Comment (MatkovIvan) [reply AppCenterCrashes/AppCenterCrashesTests/MSErrorLogFormatterTests.mm]: I've faced that `generateLiveReport` may return `nil`, and it happens sometimes on CI.

File: AppCenter/AppCenter/Internals/Sender/MSHttpSender.m [hunk 3]
```diff
@@ -351,6 +351,7 @@ - (NSURLSession *)session {
     NSURLSessionConfiguration *sessionConfiguration = [NSURLSessionConfiguration defaultSessionConfiguration];
     sessionConfiguration.timeoutIntervalForRequest = kRequestTimeout;
     _session = [NSURLSession sessionWithConfiguration:sessionConfiguration];
+    _session.delegateQueue.maxConcurrentOperationCount = 1;
```
Comment (TroubleMakerBen) [inline AppCenter/AppCenter/Internals/Sender/MSHttpSender.m]: I'm sceptical here. We're limiting to one concurrent operation and we really shouldn't do that.

File: AppCenterCrashes/AppCenterCrashesTests/MSErrorLogFormatterTests.mm (hunk 1 above)
Comment (TroubleMakerBen) [reply AppCenterCrashes/AppCenterCrashesTests/MSErrorLogFormatterTests.mm]: ok

**Comment by MatkovIvan** [comment]
Catch this! Should fail all time before, but [filename of crash file is time-based](https://github.com/Microsoft/AppCenter-SDK-Apple/blob/1.0.1/AppCenterCrashes/AppCenterCrashes/MSCrashes.mm#L898), so if multiple crashes were written in the same second, the file was overwritten.

**Comment by MatkovIvan** [comment]
🤞

**Pull request merged**